File: medium/python/11-container-with-most-water.py
class Solution:
    def maxArea(self, heights: list[int]) -> int:
        left = 0
        right = len(heights) - 1
        maxVolume = (right - left) * min(heights[right], heights[left])
        while left < right:
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
            maxVolume = max(maxVolume, (right - left) * min(heights[right], heights[left]))
        return maxVolume

File: algorithms/medium/golang/11-container-with-most-water.go
package golang

func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	area := (right - left) * min(height[right], height[left])
	for left < right {
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
		area = max(area, (right-left)*min(height[right], height[left]))
	}
	return area
}
